src/launchpad/size/treemap/macho_element_builder.py

```diff
@@ -59,220 +57,383 @@ def build_element(self, file_info: FileInfo, display_name: str) -> TreemapElemen
-        size_diff = file_info.size - total_child_size
-        size_diff_abs = abs(size_diff)
-        size_diff_percent = (size_diff_abs / file_info.size) * 100 if file_info.size > 0 else 0
-
-        logger.debug(f"Size validation for {display_name}:")
-        logger.debug(f"  File size: {file_info.size:,} bytes")
-        logger.debug(f"  Treemap total: {total_child_size:,} bytes")
-
-        if size_diff > 0:
-            logger.debug(f"  Difference: {size_diff_abs:,} bytes MISSING from treemap ({size_diff_percent:.2f}%)")
-        elif size_diff < 0:
-            logger.debug(f"  Difference: {size_diff_abs:,} bytes OVER-COUNTED in treemap ({size_diff_percent:.2f}%)")
-        else:
-            logger.debug("  Difference: 0 bytes - perfect match!")
+        total_child_size = sum(e.size for e in children)
+        diff = file_info.size - total_child_size
+        if diff != 0:
+            pct = (abs(diff) / file_info.size * 100) if file_info.size else 0
+            logger.warning(f"Size mismatch for {display_name}: file={file_info.size:,} treemap={total_child_size:,}")
+            if diff > 0:
+                logger.warning(f"  Difference: {abs(diff):,} bytes MISSING from treemap ({pct:.2f}%)")
+            else:
+                logger.warning(f"  Difference: {abs(diff):,} bytes OVER-COUNTED in treemap ({pct:.2f}%)")
 
     def _build_binary_treemap(self, binary_analysis: MachOBinaryAnalysis) -> List[TreemapElement] | None:
-        symbol_info = binary_analysis.symbol_info
-        segments = binary_analysis.segments
-
         binary_children: List[TreemapElement] = []
 
-        # Track how much of each section's bytes we "burn" while assigning
-        # bytes to symbols, so that we don't double-count them later.
+        # Section bookkeeping for remaining size
+        section_remaining: Dict[str, int] = {}
+        section_by_name: Dict[str, str] = {}  # section -> segment
+
+        for seg in binary_analysis.segments:
+            for sec in seg.sections or []:
+                key = f"{seg.name}.{sec.name}"
+                section_remaining[key] = sec.size
+                section_by_name[sec.name] = seg.name
+
+        def canonical_key(seg_name: str | None, sec_name: str | None) -> str | None:
+            if not sec_name:
+                return None
+            seg = seg_name or section_by_name.get(sec_name)
+            return f"{seg}.{sec_name}" if seg else None
+
+        def debit_section(seg_name: str | None, sec_name: str | None, sz: int) -> int:
+            if sz <= 0:
+                return 0
+            key = canonical_key(seg_name, sec_name)
+            if not key or key not in section_remaining:
+                return 0
+            take = min(sz, section_remaining[key])
+            if take:
+                section_remaining[key] -= take
+            return take
+
         section_subtractions: Dict[str, int] = {}
-        segment_subtractions: Dict[str, int] = {}
```

Comment:
These changes look a lot more involved than they actually are. Mostly I changed over to the "debit" terminology since I kept getting confused with the subtractions. And then I refactored this massive function into smaller functions since I wanted to add the "Other Symbols" node.